src/pytest_workflow/workflow_file_tests.py

```diff
@@ -0,0 +1,43 @@
+"""All tests for workflow files"""
+
+from pathlib import Path
+from typing import List
+
+import pytest
+
+
+class WorkflowFilesTestCollector(pytest.Collector):
+    def __init__(self, name, parent, files: List[dict], cwd):
+        self.files = files
+        self.name = name
+        self.cwd = cwd
+        super(WorkflowFilesTestCollector, self).__init__(name, parent=parent)
+
+    def collect(self):
+        filepaths = [Path(x.get("path")) for x in self.files]
```

Comment:
`path` must exist in the schema, right? Accessing by `x['path']` would, for illegal yaml files where `path` is not defined, at least result in a rather descriptive `KeyError` instead of the somewhat nebulous `TypeError: expected str, bytes or os.PathLike object, not NoneType` error.

Reply:
Agree. So used to getting (Scala stuff :wink: )

Reply:
;-)